homeassistant/components/binary_sensor/zha.py

```diff
@@ -54,8 +63,36 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     async_add_devices([sensor])
 
 
+@asyncio.coroutine
+def _async_setup_remote(hass, config, async_add_devices, discovery_info):
+
+    @asyncio.coroutine
+    def safe(coro):
+        """Run coro, catching ZigBee delivery errors, and ignoring them."""
+        import zigpy.exceptions
+        try:
+            yield from coro
+        except zigpy.exceptions.DeliveryError as exc:
+            _LOGGER.info("Ignoring error during setup: %s", exc)
+
+    if discovery_info['new_join']:
+        from zigpy.zcl.clusters.general import OnOff, LevelControl
+        out_clusters = discovery_info['out_clusters']
+        if OnOff.cluster_id in out_clusters:
+            cluster = out_clusters[OnOff.cluster_id]
+            yield from safe(cluster.bind())
```

Comment:
Return code from bind and configure_reporting should be evaluated and not just ignored. There is a good change a device not supports it or went to sleep. At least an error log entry would be helpful

Reply:
If I remember right, the reason this is wrapped in safe() is that some devices don't support one or the other method of setup. It's logged at info level in safe() -- I'll bump that to warning.